test/integration/overload_integration_test.cc

```diff
@@ -1,58 +1,126 @@
+  if (monitor_ == monitor) {
+    monitor_ = nullptr;
+  }
+}
+
+Server::ResourceMonitorPtr FakeResourceMonitorFactory::createResourceMonitor(
+    const Protobuf::Message&, Server::Configuration::ResourceMonitorFactoryContext& context) {
+  auto monitor = std::make_unique<FakeResourceMonitor>(context.dispatcher(), *this);
+  monitor_ = monitor.get();
+  return monitor;
+}
+
 class OverloadIntegrationTest : public HttpProtocolIntegrationTest {
 protected:
-  OverloadIntegrationTest()
-      : injected_resource_filename_(TestEnvironment::temporaryPath("injected_resource")),
-        file_updater_(injected_resource_filename_) {}
-
   void initialize() override {
-    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
-      const std::string overload_config = fmt::format(R"EOF(
+    config_helper_.addConfigModifier([](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      const std::string overload_config = R"EOF(
         refresh_interval:
           seconds: 0
           nanos: 1000000
         resource_monitors:
-          - name: "envoy.resource_monitors.injected_resource"
+          - name: "envoy.resource_monitors.testonly.fake_resource_monitor"
             typed_config:
-              "@type": type.googleapis.com/envoy.config.resource_monitor.injected_resource.v2alpha.InjectedResourceConfig
-              filename: "{}"
+              "@type": type.googleapis.com/google.protobuf.Empty
         actions:
           - name: "envoy.overload_actions.stop_accepting_requests"
             triggers:
-              - name: "envoy.resource_monitors.injected_resource"
+              - name: "envoy.resource_monitors.testonly.fake_resource_monitor"
                 threshold:
                   value: 0.9
           - name: "envoy.overload_actions.disable_http_keepalive"
             triggers:
-              - name: "envoy.resource_monitors.injected_resource"
+              - name: "envoy.resource_monitors.testonly.fake_resource_monitor"
                 threshold:
                   value: 0.8
           - name: "envoy.overload_actions.stop_accepting_connections"
             triggers:
-              - name: "envoy.resource_monitors.injected_resource"
+              - name: "envoy.resource_monitors.testonly.fake_resource_monitor"
                 threshold:
                   value: 0.95
-      )EOF",
-                                                      injected_resource_filename_);
+      )EOF";
       *bootstrap.mutable_overload_manager() =
           TestUtility::parseYaml<envoy::config::overload::v3::OverloadManager>(overload_config);
     });
     updateResource(0);
     HttpIntegrationTest::initialize();
   }
 
-  void updateResource(double pressure) { file_updater_.update(absl::StrCat(pressure)); }
+  void updateResource(double pressure) {
+    auto* monitor = fake_resource_monitor_factory_.monitor();
+    if (monitor != nullptr) {
```

Comment:
nit: Consider ASSERT_TRUE(monitor != nullptr) or ASSERT(monitor) instead to avoid having the update call ending up as a no-op if parts of the test framework are not setup properly.